rules/cusotm.yml

```diff
@@ -0,0 +1,26 @@
+# https://unicode-org.github.io/icu/userguide/boundaryanalysis/break-rules.html
+
+# Keep hyphens
+hyphen: |
+    $Hyphen = [\p{Pd}];
+    ($ALetterPlus | $Hebrew_Letter | $Numeric)  $ExFm* $Hyphen $ExFm* ($ALetterPlus | $Hebrew_Letter | $Numeric);
+
+# Keep URLs
+url: |
+    ^(([h][t][t][p][s]?\:\/\/([w][w][w]\.)?)|([w][w][w]\.))[-a-zA-Z0-9@:%_\+\.~#=]+(\/[-a-zA-Z0-9@\:%_\+\.~#?\&=]+)*[\/]?;
+
+# Keep emails
+email: |
+    [A-Za-z0-9_]+\@[A-Za-z][A-Za-z0-9_]+\.[a-z]+;
+
+# Split elisions
+elision: |
+    $Elision = ([lLmMtTnNsSjJdDcC]|([jJ][u][s]|[qQ][u][o][i]|[lL][o][r][s]|[pP][u][i][s]|[qQ][u][e][l])?[qQ][u])[\u0027\u2019];
+    ^$Elision / $ALetterPlus;
+
+
+# Split tags
+tag: |
+    [#] / [\\p{L}\\p{N}]+; 
+    [\\p{L}\\p{N}]+ / [#]; 
+    [@] / [a-zA-Z0-9_]+;
```

Comment:
I would advise using a leading `^` to match only the beginning of a word. Otherwise, username such as koheiw@quanteda would be affected as well.

Reply:
Update!